deploy/lib/createFunctions.js

```diff
@@ -53,49 +54,13 @@ module.exports = {
     });
   },
 
-  applyDomains(funcId, customDomains) {
+  applyDomainsFunc(funcId, customDomains) {
     // we make a diff to know which domains to add or delete
-    const domainsToCreate = [];
-    const domainsIdToDelete = [];
-    const existingDomains = [];
 
-    this.listDomains(funcId).then((domains) => {
-      domains.forEach((domain) => {
-        existingDomains.push({ hostname: domain.hostname, id: domain.id });
-      });
-
-      if (
-        customDomains !== undefined &&
-        customDomains !== null &&
-        customDomains.length > 0
-      ) {
-        customDomains.forEach((customDomain) => {
-          domainsIdToDelete.push(customDomain.id);
-
-          let domainFound = false;
-
-          existingDomains.forEach((existingDom) => {
-            if (existingDom.hostname === customDomain) {
-              domainFound = true;
-              return false;
-            }
-          });
-          if (!domainFound) {
-            domainsToCreate.push(customDomain);
-          }
-        });
-      }
-
-      existingDomains.forEach((existingDomain) => {
-        if (
-          (customDomains === undefined || customDomains === null) &&
-          existingDomain.id !== undefined
-        ) {
-          domainsIdToDelete.push(existingDomain.id);
-        } else if (!customDomains.includes(existingDomain.hostname)) {
-          domainsIdToDelete.push(existingDomain.id);
-        }
-      });
+    this.listDomainsFunction(funcId).then((domains) => {
```

Comment:
Is the block after `(domains) => {` the same as in the `applyDomainsContainer`. If so, can you extract the code to a function and reuse it?